code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -72,10 +82,18 @@
 
 	finish_charging()
 	playsound(owner, 'sound/effects/petrify_activate.ogg', 50)
-	for(var/mob/living/carbon/human/human in view(PETRIFY_RANGE, owner.loc))
-		if(is_blind(human))
+	for(var/mob/living/carbon/carbon_viewer in view(PETRIFY_RANGE, owner.loc))
+		if(isxeno(carbon_viewer))
+			if(!petrify_armor)
+				continue
+			var/mob/living/carbon/xenomorph/xenomorph_viewer = carbon_viewer
+			var/datum/armor/attaching_armor = getArmor()
```

Comment:
empty getarmor? also just realized a free performance optimization, the above can use the ssspatial get hearers in view if you feel like doing it